firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/firestore.kt

```diff
@@ -80,35 +74,85 @@ expect open class Query {
+
+@Deprecated("Deprecated in favor of using a [FilterBuilder]", replaceWith = ReplaceWith("where { field equalTo equalTo }", "dev.gitlive.firebase.firestore"))
+fun Query.where(field: String, equalTo: Any?) = where {
+    field equalTo equalTo
 }
 
-fun Query.where(field: String, equalTo: Any?) = _where(field, equalTo?.value)
-fun Query.where(path: FieldPath, equalTo: Any?) = _where(path, equalTo?.value)
-fun Query.where(field: String, equalTo: DocumentReference) = _where(field, equalTo.value)
-fun Query.where(path: FieldPath, equalTo: DocumentReference) = _where(path, equalTo.value)
-fun Query.where(field: String, lessThan: Any? = null, greaterThan: Any? = null, arrayContains: Any? = null) = _where(field, lessThan?.value, greaterThan?.value, arrayContains?.value)
-fun Query.where(path: FieldPath, lessThan: Any? = null, greaterThan: Any? = null, arrayContains: Any? = null) = _where(path, lessThan?.value, greaterThan?.value, arrayContains?.value)
-fun Query.where(field: String, inArray: List<Any>? = null, arrayContainsAny: List<Any>? = null) = _where(field, inArray?.value, arrayContainsAny?.value)
-fun Query.where(path: FieldPath, inArray: List<Any>? = null, arrayContainsAny: List<Any>? = null) = _where(path, inArray?.value, arrayContainsAny?.value)
+@Deprecated("Deprecated in favor of using a [FilterBuilder]", replaceWith = ReplaceWith("where { path equalTo equalTo }", "dev.gitlive.firebase.firestore"))
+fun Query.where(path: FieldPath, equalTo: Any?) = where {
+    path equalTo equalTo
+}
+
+@Deprecated("Deprecated in favor of using a [FilterBuilder]", replaceWith = ReplaceWith("where {  }", "dev.gitlive.firebase.firestore"))
+fun Query.where(field: String, lessThan: Any? = null, greaterThan: Any? = null, arrayContains: Any? = null) = where {
+    val filters = listOfNotNull(
+        lessThan?.let { field lessThan it },
+        greaterThan?.let { field greaterThan it },
+        arrayContains?.let { field contains it }
+    )
+    filters.fold<Filter, Filter?>(null) { acc, filter ->
+       acc?.let { it and filter } ?: filter
+    }
+}
+
+@Deprecated("Deprecated in favor of using a [FilterBuilder]", replaceWith = ReplaceWith("where {  }", "dev.gitlive.firebase.firestore"))
+fun Query.where(path: FieldPath, lessThan: Any? = null, greaterThan: Any? = null, arrayContains: Any? = null) = where {
+    val filters = listOfNotNull(
+        lessThan?.let { path lessThan it },
+        greaterThan?.let { path greaterThan it },
+        arrayContains?.let { path contains it }
+    )
+    filters.fold<Filter, Filter?>(null) { acc, filter ->
+        acc?.let { it and filter } ?: filter
+    }
+}
+
+@Deprecated("Deprecated in favor of using a [FilterBuilder]", replaceWith = ReplaceWith("where {  }", "dev.gitlive.firebase.firestore"))
+fun Query.where(field: String, inArray: List<Any>? = null, arrayContainsAny: List<Any>? = null) = where {
+    val filters = listOfNotNull(
+        inArray?.let { field `in` it },
```

Comment:
Could you overload the in operator to avoid the quotes?

https://kotlinlang.org/docs/operator-overloading.html#in-operator

Reply:
I cant overload it since the method to overload is contains which has a different signature. Ill rename to inArray

Reply:
For similar reasons I cant make operator overloads for < > etc. They all go to the compareTo method and that isnt suitable for overloading in this case